zilliqa/src/api/types/eth.rs

```diff
@@ -158,6 +159,67 @@ pub struct Transaction {
     pub transaction_type: u64,
 }
 
+impl Transaction {
+    pub fn new(tx: transaction::VerifiedTransaction, block: Option<message::Block>) -> Self {
+        let hash = tx.hash;
+        let from = tx.signer;
+        let v = tx.tx.sig_v();
+        let r = tx.tx.sig_r();
+        let s = tx.tx.sig_s();
+        let transaction = tx.tx.into_transaction();
+        let (gas_price, max_fee_per_gas, max_priority_fee_per_gas) = match transaction {
+            transaction::Transaction::Legacy(_)
+            | transaction::Transaction::Eip2930(_)
+            | transaction::Transaction::Zilliqa(_)
+            | transaction::Transaction::Intershard(_) => {
+                (transaction.max_fee_per_gas(), None, None)
+            }
+            transaction::Transaction::Eip1559(TxEip1559 {
+                max_fee_per_gas,
+                max_priority_fee_per_gas,
+                ..
+            }) => (
+                // The `gasPrice` for EIP-1559 transactions should be set to the effective gas price of this transaction,
+                // which depends on the block's base fee. We don't yet have a base fee so we just set it to the max fee
+                // per gas.
```

Comment:
Worth an issue if we're going to have a base fee in the future, or this will turn up as a bug when we do ..

Reply:
#962